stl/inc/atomic

```diff
@@ -125,6 +125,38 @@ _NODISCARD extern "C" bool __cdecl __std_atomic_has_cmpxchg16b() noexcept;
 #endif
 
 _STD_BEGIN
+// STRUCT TEMPLATE _Storage_for
+struct _Form_mask_t {};
+_INLINE_VAR constexpr _Form_mask_t _Form_mask;
+
+template <class _Ty>
+struct _Storage_for {
+    // uninitialized space to store a _Ty
+    alignas(_Ty) unsigned char _Storage[sizeof(_Ty)];
+
+    _Storage_for()                    = default;
+    _Storage_for(const _Storage_for&) = delete;
+    _Storage_for& operator=(const _Storage_for&) = delete;
+
+    explicit _Storage_for(_Form_mask_t) noexcept {
+        _CSTD memset(_Storage, 0xff, sizeof(_Ty));
+        __builtin_zero_non_value_bits(_Ptr());
+    }
+
+    _Ty& _Ref() noexcept {
+        return reinterpret_cast<_Ty&>(_Storage);
+    }
+
+    _Ty* _Ptr() noexcept {
+        return reinterpret_cast<_Ty*>(_STD addressof(_Storage));
+    }
+};
+
+#if _HAS_CXX20

```

Comment:
Should this instead be guarded by `#if _CMPXCHG_MASK_OUT_PADDING_BITS` which appears to surround every use?